UI/platform-osx.mm

```diff
@@ -62,10 +62,13 @@ string GetDefaultVideoSavePath()
     NSURL *url = [fm URLForDirectory:NSMoviesDirectory inDomain:NSUserDomainMask appropriateForURL:nil create:true
                                error:nil];
 
-    if (!url)
-        return getenv("HOME");
+    if (!url) {
+        NSString *path = [NSString pathWithComponents:@[getenv("HOME"), @"/OBS Studio"]];
```

Comment:
```suggestion
        NSString *path = [NSString pathWithComponents:@[@(getenv("HOME")), @"OBS-Studio"]];
```

The error is "collection element of type 'char *' is not an Objective-C object" - `pathWithComponents` takes an `NSArray`, but that class requires that the type of values in the array itself are Foundation classes. `getenv("HOME")` returns a `char *`, which is not a Foundation type, so it needs to be boxed with another `@()` to convert it into an `NSString`. The danger about this is that the program will crash if `getenv` returns a NULL pointer.

Also `pathWithComponents` already concatenates with a path separator between each component, so the forward slash in the second string is superfluous.